Add a summary field to alerts

MrZammler opened the pull request:
##### Summary
<!--
Describe the change in summary section, including rationale and design decisions.
Include "Fixes #nnn" if you are fixing an existing issue.
-->

Adds a `summary` field to alerts. This is a short text field that can be used in some places where we currently use the alert name.

The field can be enriched as with the info field, with chart labels. This brings some more information in a glance in e.g. notifications subjects and our alert list in dashboard.

As an example, the same notification will change from:

![image](https://github.com/netdata/netdata/assets/1905463/d223b352-b271-45c1-86da-8aa9141b3e60)

to

![image](https://github.com/netdata/netdata/assets/1905463/ed10dbf8-461c-4601-b7cc-8be2dd119468)

Because we plan to remove the family entry from alerts, we need a way to present alert specific instance information to notifications.

##### Test Plan

<!--
Provide enough detail so that your reviewer can understand which test cases you
have covered, and recreate them if necessary. If our CI covers sufficient tests, then state which tests cover the change.
-->

##### Additional Information
<!-- This is usually used to help others understand your
motivation behind this change. A step-by-step reproduction of the problem is
helpful if there is no related issue. -->

<details> <summary>For users: How does this change affect me?</summary>
  <!--
Describe the PR affects users: 
- Which area of Netdata is affected by the change?
- Can they see the change or is it an under the hood? If they can see it, where?
- How is the user impacted by the change? 
- What are there any benefits of the change? 
-->
</details>

sashwathn commented:
@hugovalente-pm : I don't see this impacting the Mobile App in anyway if it is purely enriching the Alert Information.. Once the payload includes this, the app will automatically use this - unless we set them up as a separate field.
@car12o Any comments?

car12o commented:
> @hugovalente-pm : I don't see this impacting the Mobile App in anyway if it is purely enriching the Alert Information.. Once the payload includes this, the app will automatically use this - unless we set them up as a separate field. @car12o Any comments?

I'm not yet fully aware of this feature but from glance I think it will be a separate field.

hugovalente-pm commented:
@car12o @sashwathn yes this is a new filed, the initial list of task (to be reviewed/confirmed) is stating that https://github.com/netdata/netdata-cloud/issues/130#issuecomment-1699276801 since we are not getting rid of the previous `alert name` we are inrtoducing this `summary` field

MrZammler commented:
I've added the summary field to some alerts. More can be added later (assuming we're ok with what I've added so far).

When this is merged, we then need another PR to transmit it to the cloud and then change our own and cloud backend's endpoints to provide it to the FE.

We also need to add it to other notifications besides email.

thiagoftsm commented:
Hello @MrZammler ,

Please, take a look in Codacy warning https://app.codacy.com/gh/netdata/netdata/pullRequest?prid=12502044 .

Best regards!

thiagoftsm commented on database/sqlite/sqlite_db_migration.c:
```diff
@@ -88,6 +88,12 @@ const char *database_migrate_v10_v11[] = {
 };
 
+const char *database_migrate_v11_v12[] = {
+    "ALTER TABLE health_log_detail ADD summary TEXT;",
```
When we work with databases like PostgreSQL and MySQL/MariaDB in normal situation it is good to add collation for field to accept UTF-8, because `TEXT` means for these databases only `ASCII` pattern. Can our users to use UTF-8 summaries? Or are we going to have only `ASCII`?

stelfrag replied:
Not a problem -- from the sqlite docs `TEXT. The value is a text string, stored using the database encoding (UTF-8, UTF-16BE or UTF-16LE).`

thiagoftsm commented on database/sqlite/sqlite_db_migration.c:
```diff
@@ -315,6 +321,25 @@ static int do_migration_v10_v11(sqlite3 *database, const char *name)
 
+static int do_migration_v11_v12(sqlite3 *database, const char *name)
+{
+    char sql[2048];
```
Your query does not use 128 bytes, why are you using 2048 bytes here?

thiagoftsm commented on database/sqlite/sqlite_db_migration.c:
```diff
@@ -315,6 +321,25 @@ static int do_migration_v10_v11(sqlite3 *database, const char *name)
+        rc = init_database_batch(database, &database_migrate_v11_v12[0]);
+
+    if (!rc) {
+        snprintfz(sql, 2047, "UPDATE health_log_detail SET summary = (select name from health_log where health_log_id = health_log_detail.health_log_id);");
```
Is really necessary to copy a string to a vector? Can we use only a `define` here?

stelfrag commented on database/sqlite/sqlite_health.c:
```diff
@@ -891,6 +896,11 @@ void sql_health_alarm_log_load(RRDHOST *host)
 
         ae->chart_name = SQLITE3_COLUMN_STRINGDUP_OR_NULL(res, 33);
 
+        if (sqlite3_column_type(res, 34) != SQLITE_NULL)
```
This can be 
```suggestion
        ae->summary = SQLITE3_COLUMN_STRINGDUP_OR_NULL(res, 34);
```

stelfrag commented on database/sqlite/sqlite_health.c:
```diff
@@ -891,6 +896,11 @@ void sql_health_alarm_log_load(RRDHOST *host)
         ae->chart_name = SQLITE3_COLUMN_STRINGDUP_OR_NULL(res, 33);
 
+        if (sqlite3_column_type(res, 34) != SQLITE_NULL)
+            ae->summary = string_strdupz((char *) sqlite3_column_text(res, 34));
```
```suggestion
```

stelfrag commented on database/sqlite/sqlite_health.c:
```diff
@@ -891,6 +896,11 @@ void sql_health_alarm_log_load(RRDHOST *host)
 
+        if (sqlite3_column_type(res, 34) != SQLITE_NULL)
+            ae->summary = string_strdupz((char *) sqlite3_column_text(res, 34));
+        else
```
```suggestion
```

MrZammler commented:
Thanks guys, I think last commit resolved the comments.

thiagoftsm reviewed (approved):
PR is working as expected, LGTM!
